### PR TITLE
Generate doc for Factory#load and Factory#dump

### PR DESCRIPTION
**Pull request by casperisfine**
Doc update for https://github.com/msgpack/msgpack-ruby/pull/142 as requested.

cc @tagomoris

### CODE REVIEW

**Comment by tagomoris** [comment]
@casperisfine Ah, I'm sorry that I missed what I should tell you...
The API documents of this library are written under `doclib` in `master` (and other development) branch. So the exact list of what we need to do when we want to update API document to add new methods is (for example, add new methods to `MessagePack::Factory`):

1. update `doclib/msgpack/factory.rb` to add methods
2. create a pull-req to master branch with the diff above
3. execute `bundle exec rake doc` with the diff above to generate updated documents under `doc`
4. make a pull-req to `gh-pages` with contents under `doc` (by copying it to root)

**Comment by casperisfine** [comment]
I'll get right on that!

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@ Gemfile*
 *~
 /msgpack-ruby
 /vendor
+/.yardoc
+/doc
```
Comment (casperisfine) [inline .gitignore]: I took the liberty to add those 2, it makes the process much simpler.

**Comment by casperisfine** [comment]
Weird. I ran `rake doc` from https://github.com/msgpack/msgpack-ruby/pull/145, but it doesn't seem to generate doc for the methods I added 🤔

**Comment by tagomoris** [comment]
I'm supposing that it's caused by this blank line: https://github.com/msgpack/msgpack-ruby/pull/145#discussion_r155202275

**Comment by casperisfine** [comment]
I fixed it, but still no luck :/

**Comment by tagomoris** [comment]
I created #146 and merged it because it seems to work well.
@casperisfine  Thank you!

**Pull request closed without merging**